### PR TITLE
fix(owl-bot): restrict organization for running post processor

### LINKED ISSUE #2750: Owlbot is installed on odd external repository
OwlBot is installed on a strange external dependency:

```
 Error creating label owlbot:ignore in QPC-WORLDWIDE/bioconda-repodata-archive-old Repository is above its size quota.
Error creating label owlbot:ignore in QPC-WORLDWIDE/bioconda-repodata-archive-old Repository is above its size quota.
```

This appears to be interfering with our ability to complete the label sync job successfully.

### PR DESCRIPTION
**Pull request by tmatsuo**
fixes #2750

### CODE REVIEW

File: packages/owl-bot/test/owl-bot.ts [hunk 1]
```diff
@@ -130,6 +130,38 @@ describe('owlBot', () => {
...
           },
+          base: {
+            ref: 'main',
+            repo: {
+              full_name: 'bcoe/owl-bot-testing',
+            },
+          },
+        },
+      };
+      const loggerStub = sandbox.stub(logger, 'info');
```
Comment (bcoe) [inline packages/owl-bot/test/owl-bot.ts:141]: I like your approach of stubbing the logger to test this.

File: packages/owl-bot/src/owl-bot.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ import {octokitFactoryFrom} from './octokit-util';
 import {REGENERATE_CHECKBOX_TEXT} from './copy-code';
 import {githubRepo} from './github-repo';
 
-const SYNC_LABEL_ORGANIZATIONS = ['googleapis', 'GoogleCloudPlatform'];
+const ALLOWED_ORGANIZATIONS = ['googleapis', 'GoogleCloudPlatform'];
```
Comment (bcoe) [inline packages/owl-bot/src/owl-bot.ts:37]: GitHub orgs are not case sensitive, we might be smart to use lowercase names for organizations and then call `.toLowerCase()` on the payload. Just in case we bump into a weird edge-case.

File: packages/owl-bot/test/owl-bot.ts [hunk 3]
```diff
@@ -1601,23 +1633,23 @@ describe('owlBot', () => {
...
         },
       },
     };
     const githubMock = nock('https://api.github.com')
-      .get('/repos/bcoe/owl-bot-testing/pulls/33')
+      .get('/repos/googleapis/owl-bot-testing/pulls/33')
       .reply(200, payload.pull_request)
       .get(
-        '/repos/bcoe/owl-bot-testing/contents/.github%2F.OwlBot.lock.yaml?ref=abc123'
+        '/repos/googleapis/owl-bot-testing/contents/.github%2F.OwlBot.lock.yaml?ref=abc123'
```
Comment (bcoe) [inline packages/owl-bot/test/owl-bot.ts:1652]: maybe add one test with `googlecloudplatform`, just to test both cases.

**Review by bcoe — CHANGES_REQUESTED** [review]
nit: I think it might be worth considering making the check for org case-insenstive.

**Review by SurferJeffAtGoogle — APPROVED** [review]

File: packages/owl-bot/src/owl-bot.ts (hunk 2 above)
Comment (tmatsuo) [reply packages/owl-bot/src/owl-bot.ts:37]: done

File: packages/owl-bot/test/owl-bot.ts (hunk 3 above)
Comment (tmatsuo) [reply packages/owl-bot/test/owl-bot.ts:1652]: Added one test with `GoogleCloudPlatform` (CamelCased)

**Comment by tmatsuo** [comment]
@bcoe PTAL

**Review by bcoe — APPROVED** [review]
